### ISSUE TITLE
Improve nuget.org's Markdown support 

### ISSUE THREAD

**Issue by loic-sharma** [NuGet/NuGetGallery#8837]
Since .NET 5.0, you can [add a README to your NuGet package](https://devblogs.microsoft.com/nuget/add-a-readme-to-your-nuget-package/). Here are a few Markdown features we should add to nuget.org to make this experience amazing:

* [ ] https://github.com/NuGet/NuGetGallery/issues/8503
* [ ] https://github.com/NuGet/NuGetGallery/issues/8505
* [ ] https://github.com/NuGet/NuGetGallery/issues/8632
* [ ] https://github.com/NuGet/NuGetGallery/issues/8644
* [ ] https://github.com/NuGet/NuGetGallery/issues/8726
* [ ] https://github.com/NuGet/NuGetGallery/issues/8802
* [ ] https://github.com/NuGet/NuGetGallery/issues/8833

**Comment by chgill-MSFT** [comment]
From Loic's README analysis for .NET repos with over 100 stars:

| Issue                    	| % impacted 	|
|--------------------------	|------------	|
| Syntax highlighting      	| 80%        	|
| Additional image domains 	| 71%        	|
| HTML support             	| 29%        	|
| Table rendering          	| 27%        	|
| Mailto links             	| 9%         	|
| Code snippet copy button 	|            	|
| Section links            	|            	|
| Blockquote rendering     	|            	|

@loic-sharma could we find out what percentage are impact by the last 3 items?

**Comment by lyndaidaii** [comment]
@chgill-MSFT  Agree,  I think for the HTML we might want to break it down to more specific like header, image, css and so on, how about gather more information/analysis to make it more in details? 
Also we could just fix table render, blockquotes since it's small engineering cost with better customer experience.
